libminifi/src/sitetosite/RawSocketProtocol.cpp

```diff
@@ -310,31 +310,32 @@ bool RawSiteToSiteClient::handShake() {
     }
   }
 
-  std::string error;
+  auto logPortStateError = [this](const std::string& error) {
+    logger_->log_error("Site2Site HandShake Failed because destination port, %s, is %s", port_id_.to_string(), error);
+  };
 
   switch (code) {
     case PROPERTIES_OK:
       logger_->log_debug("Site2Site HandShake Completed");
       peer_state_ = HANDSHAKED;
       return true;
     case PORT_NOT_IN_VALID_STATE:
-      error = "in invalid state";
-      break;
+      logPortStateError("in invalid state");
+      return false;
     case UNKNOWN_PORT:
-      error = "an unknown port";
-      break;
+      logPortStateError("an unknown port");
+      return false;
     case PORTS_DESTINATION_FULL:
-      error = "full";
-      break;
+      logPortStateError("full");
+      return false;
+    case UNAUTHORIZED:
+      logger_->log_error("Site2Site HandShake failed: UNAUTHORIZED");
+      return false;
     // Unknown error
     default:
       logger_->log_error("HandShake Failed because of unknown respond code %d", code);
       return false;
```

Comment:
these two could use `logPortStateError()`, too: knowing the port number could be useful in these cases, too

also, I would remove the comment in line 334

Reply:
as discussed, we can leave it like this for now